Rover/Log.cpp

```diff
@@ -38,44 +38,52 @@ void Rover::Log_Write_Attitude()
 // Write a range finder depth message
 void Rover::Log_Write_Depth()
 {
-    // only log depth on boats with working downward facing range finders
-    if (!rover.is_boat() || !rangefinder.has_data_orient(ROTATION_PITCH_270)) {
+    // only log depth on boats
+    if (!rover.is_boat()) {
```

Comment:
Looks like were still only using `ROTATION_PITCH_270` so we could still early return?

Reply:
```suggestion
    if (!rover.is_boat() || !rangefinder.has_orientation(ROTATION_PITCH_270)) {
```
Would need to swap from `has_data_orient` to `has_orientation`

Reply:
This would be looping through all of the backends max. 2 extra times just for find-instance() in has_orientation() check.  Though it does look prettier. 

Below loops only num_instances after your suggestion ;)